test/CMakeLists.txt

```diff
@@ -177,3 +184,25 @@ add_umf_test(NAME base_alloc_linear
 add_umf_test(NAME base_alloc_global
             SRCS ${BA_SOURCES_FOR_TEST} pools/pool_base_alloc.cpp malloc_compliance_tests.cpp
             LIBS umf_utils)
+
+add_umf_test(NAME pool_os_alloc
+            SRCS pools/pool_os_alloc.cpp malloc_compliance_tests.cpp)
+
+# tests for the proxy library
+if((LINUX OR WINDOWS) AND UMF_BUILD_OS_MEMORY_PROVIDER AND UMF_BUILD_SHARED_LIBRARY)
+    if(WINDOWS AND NOT (CMAKE_BUILD_TYPE STREQUAL Release))
+        message(STATUS "Skipping the tests of the proxy library, because they are supported only in the Release build on Windows")
+    elseif((UMF_PROXY_LIB_BASED_ON_POOL STREQUAL SCALABLE) AND (NOT UMF_BUILD_LIBUMF_POOL_SCALABLE))
+        message(STATUS "Skipping the tests of the proxy library, because UMF_PROXY_LIB_BASED_ON_POOL==SCALABLE but UMF_BUILD_LIBUMF_POOL_SCALABLE is OFF")
+    elseif((UMF_PROXY_LIB_BASED_ON_POOL STREQUAL JEMALLOC) AND (NOT UMF_BUILD_LIBUMF_POOL_JEMALLOC))
```

Comment:
Wouldn't it be better to just set some var UMF_SCALABLE_POOL_ENABLED in the main CMake after validating all the options and just have a single if() here?

Reply:
Right. Done.